contracts/transfer/src/state.rs

```diff
@@ -343,6 +465,12 @@ impl TransferState {
         }
     }
 
+    /// Return the current gas price as set by the execute method.
+    /// Returns none outside of the lifetime of the execute method.
+    pub fn gas_price(&self) -> Option<u64> {
+        self.gas_price
+    }
```

Comment:
It doesn't seem to me that there is an "outside the lifetime of the execute method." It is literally the entrypoint of the network's state. The only time that we do not enter via execute is during queries, and at that point this could just panic, so I suggest an expect:

```suggestion
    /// Return the current gas price as set by the execute method.
    /// Returns none outside of the lifetime of the execute method.
    pub fn gas_price(&self) -> u64 {
        self.gas_price.expect("During transaction execution the host should always set the gas price")
    }
```

Reply:
sounds good, I like this suggestion